client/go/internal/vespa/target.go

```diff
@@ -221,16 +223,26 @@ func (s *Service) Description() string {
 }
 
 // FindService returns the service of given name, found among services, if any.
-func FindService(name string, services []*Service) (*Service, error) {
-	if name == "" && len(services) == 1 {
-		return services[0], nil
+func FindService(name string, authMethod string, services []*Service) (*Service, error) {
+	applicableServices := make([]*Service, 0, len(services))
+	for _, s := range services {
+		if s.AuthMethod == authMethod || s.AuthMethod == "" {
+			applicableServices = append(applicableServices, s)
+		}
+	}
+	if name == "" && len(applicableServices) == 1 {
+		return applicableServices[0], nil
 	}
-	names := make([]string, len(services))
-	for i, s := range services {
+	names := make([]string, 0, len(services))
+	for _, s := range services {
 		if name == s.Name {
 			return s, nil
 		}
-		names[i] = s.Name
+		var prettyName = color.CyanString("%s", s.Name)
```

Comment:
When adding ANSI color we should also add an option to print colourless, not all shells support ansi colors.

Furthermore: if we use colors, we should define a system and apply thought out the whole messages, not just "in one place". 😊

Reply:
👍🏼  Right, I didn't realize that was already a dependency, apologies. 😊